host/bcm283x/dma.go

```diff
@@ -412,6 +412,8 @@ type controlBlock struct {
 	reserved [2]uint32 // 0x18+0x1C
 }
 
+const controlBlockSize uint32 = 32
```

Comment:
I'd recommend removing the uint32 type, as then it'll be used as the correct type in context, which will remove the need for casting as you did at line 901.

Thanks for adding one, using 32 everywhere was a bit dumb.

Reply:
well there's offsetBytes that would be incorrect. So you'd save two casts but add a new one. I'd still vote for removing since it's customary in Go.